src/combinatorics/simple_graph/adj_matrix.lean

```diff
@@ -15,89 +16,247 @@ properties to computational properties of the matrix.
+  (3) every diagonal entry of `A` is `0`.
+
+* `matrix.is_adj_matrix.to_graph`: for `A : matrix V V α` and `h : A.is_adj_matrix`,
+  `h.to_graph` is the simple graph induced by `A`.
+
+* `matrix.compl`: for `A : matrix V V α`, `A.compl` is supposed to be
+  the adjacency matrix of the complement graph of the graph induced by `A`.
+
+* `simple_graph.adj_matrix`: the adjacency matrix of a `simple_graph`.
 
 -/
 
 open_locale big_operators matrix
 open finset matrix simple_graph
 
-universes u v
-variables {α : Type u}
-variables (R : Type v) [semiring R]
+variables {V α β : Type*}
+
+namespace matrix
+
+/-- `A : matrix V V α` is qualified as an "adjacency matrix" if
+    (1) every entry of `A` is `0` or `1`,
+    (2) `A` is symmetric,
+    (3) every diagonal entry of `A` is `0`. -/
+structure is_adj_matrix [has_zero α] [has_one α] (A : matrix V V α) : Prop :=
+(zero_or_one : ∀ i j, (A i j) = 0 ∨ (A i j) = 1 . obviously)
+(symm : A.is_symm . obviously)
+(loopless : ∀ i, A i i = 0 . obviously)
+
+namespace is_adj_matrix
+
+variables {A : matrix V V α}
+
+lemma apply_diag [has_zero α] [has_one α] (h : is_adj_matrix A) (i : V) :
+  A i i = 0 :=
+by simp [h.loopless]
+
+@[simp]
+lemma apply_diag_ne [mul_zero_one_class α] [nontrivial α] (h : is_adj_matrix A) (i : V) :
+  ¬ A i i = 1 :=
+by simp [h.apply_diag i]
+
+@[simp]
+lemma apply_ne_one_iff [mul_zero_one_class α] [nontrivial α] (h : is_adj_matrix A) (i j : V) :
+  ¬ A i j = 1 ↔ A i j = 0 :=
+begin
+  replace h := h.zero_or_one i j,
+  split,
+  { tauto },
+  { rintros g, simp [g] }
+end
+
+@[simp]
+lemma apply_ne_zero_iff [mul_zero_one_class α] [nontrivial α] (h : is_adj_matrix A) (i j : V) :
+  ¬ A i j = 0 ↔ A i j = 1 :=
+by { rw [←apply_ne_one_iff h], simp }
+
+/-- For `A : matrix V V α` and `h : is_adj_matrix A`,
+    `h.to_graph` is the simple graph whose adjacency matrix is `A`. -/
+def to_graph [mul_zero_one_class α] [nontrivial α] (h : is_adj_matrix A) :
+  simple_graph V :=
+{ adj := λ i j, A i j = 1,
+  sym := λ i j hij, by simp only [h.symm.apply i j]; convert hij,
+  loopless := λ i, by simp [h] }
+
+instance [mul_zero_one_class α] [nontrivial α] [decidable_eq α] (h : is_adj_matrix A) :
+  decidable_rel h.to_graph.adj :=
+by {simp [to_graph], apply_instance}
+
+end is_adj_matrix
+
+/-- For `A : matrix V V α`, `A.compl` is supposed to be the adjacency matrix of
+    the complement graph of the graph induced by `A.adj_matrix`. -/
+def compl [has_zero α] [has_one α] [decidable_eq α] [decidable_eq V] (A : matrix V V α) :
+  matrix V V α :=
+λ i j, ite (i = j) 0 (ite (A i j = 0) 1 0)
+
+section compl
+
+variables [decidable_eq α] [decidable_eq V] (A : matrix V V α)
+
+@[simp]
+lemma compl_apply_diag [has_zero α] [has_one α] (i : V) :
+  A.compl i i = 0 :=
+by simp [compl]
+
+@[simp]
+lemma compl_apply [has_zero α] [has_one α] (i j : V) :
+  A.compl i j = 0 ∨ A.compl i j = 1 :=
+begin
+  unfold compl,
+  by_cases h : i = j, {simp [h]},
+  by_cases g : A i j = 0; simp [h , g]
+end
```

Comment:
```suggestion
by { unfold compl, split_ifs; simp, }
```

I'm only writing this as a one-liner since it's the proof is "trivial."  It's also fine using `begin`/`end` as a two-line proof:
```lean
begin
  unfold compl,
  split_ifs; simp,
end
```